doc/source/release/2.0.0-notes.rst

```diff
@@ -4,6 +4,93 @@
 NumPy 2.0.0 Release Notes
 =========================
 
+NumPy 2.0.0 is the first major release since 2006. It is the result of X months
+of development since the last feature release by Y contributors, and contains a
+large amount of exciting new features as well as a large amount of changes to
+both the Python and C APIs.
+
+Key documents related to how to adapt to changes in NumPy 2.0, in addition to
+these release notes, include:
+
+- The :ref:`numpy-2-migration-guide`
+- The :ref:`NumPy 2.0-specific advice <numpy-2-abi-handling>` in
+  :ref:`for-downstream-package-authors`
+
+
+Highlights
+==========
+
+Highlights of this release include:
+
+- New features:
+
+  - A new variable-length string dtype, `~numpy.dtypes.StringDType` and a new
+    `numpy.strings` namespace with performant ufuncs ...
+  - Support for ``float32`` and ``longdouble`` in all `numpy.fft` functions,
+  - Support for the array API standard in the main ``numpy`` namespace
+
+- Performance improvements:
+
+  - Sorting functions (`sort`, `argsort`, `partition`, `argpartition`)
+    have been accelerated through the use of the Intel x86-simd-sort and Google
+    Highway libraries, and may see large (hardware-specific) speedups,
+  - macOS Accelerate support and binary wheels for macOS >=14, with significant
+    performance improvements for linear algebra operations on macOS
+  - `numpy.char` fixed-length string dtypes have been accelerated, ...
+  - A new tracing and introspection API to determine which hardware-specific
+    kernels are available and will be dispatched to ... see
+    `~numpy.lib.introspect.opt_func_info`
+
+- Python API improvements:
+
+  - A clear split between public and private API, with a new
+    :ref:`module structure <module-structure>`, and each public function now
+    available in a single place,
+  - Many removals of non-recommended functions and aliases. This should make
+    it easier to learn and use NumPy. The number of objects in the main
+    namespace decreased by ~10% and in ``numpy.lib`` by ~80%,
+  - :ref:`Canonical dtype names <canonical-python-and-c-types>` and a new
+    `~numpy.isdtype` introspection function,
+
+- C API improvements:
+
+  - A new public C API for creating custom dtypes
+  - Many outdated functions and macros removed, and private internals hidden to
+    ease future extensibility,
+  - New, easier to use, initialization functions,
```

Comment:
Is it standard to include trailing commas in list items?

Reply:
It's common; there's no style guide with this level of detail. This doc requires a full copy-edit later on, so I will resolve this for now.